### PR TITLE
correct enum usage and decide enum generation according to usage

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/2832

- correct usage of enum
- Resolve enum type according to the usage

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs [hunk 1]
```diff
@@ -380,7 +380,7 @@ private static InputType CreateType(Schema schema, string? format, IReadOnlyDict
             Namespace: schema.Extensions?.Namespace,
             Accessibility: schema.Extensions?.Accessibility,
             Description: schema.CreateDescription(),
-            Usage: InputModelTypeUsage.RoundTrip,
+            Usage: InputModelTypeUsage.None,
```
Comment (archerzz) [inline src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs:383]: why do we change to `None`?

File: src/CADL.Extension/Emitter.Csharp/src/emitter.ts [hunk 2]
```diff
@@ -450,11 +450,7 @@ function setUsage(
         } else if (usages.roundTrips.includes(name)) {
             m.Usage = Usage.RoundTrip;
         } else {
-            if ((m as InputEnumType).IsExtensible) {
-                m.Usage = Usage.RoundTrip;
-            } else {
-                m.Usage = Usage.None;
-            }
+            m.Usage = Usage.None;
```
Comment (archerzz) [inline src/CADL.Extension/Emitter.Csharp/src/emitter.ts:453]: I think we need to report to CADL team to have a usage calculation here.

File: src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs (hunk 1 above)
Comment (chunyu3) [reply src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs:383]: It is from swagger, and there is no need to generate enum, so the Usage is None

File: src/CADL.Extension/Emitter.Csharp/src/emitter.ts (hunk 2 above)
Comment (chunyu3) [reply src/CADL.Extension/Emitter.Csharp/src/emitter.ts:453]: No. cadl compiler correctly resolve the usage for enum. We used to hard-code extensible enum to RoundTrip as workround. Now we will resolve the enum issue, and this workround need to be removed.

**Review by lirenhe — COMMENTED** [review]
I didn't see any tests updated, so we don't have any coverage for this part?
Could you add tests to verify the behaviour?

**Review by archerzz — APPROVED** [review]

**Comment by chunyu3** [comment]
> I didn't see any tests updated, so we don't have any coverage for this part? Could you add tests to verify the behaviour?

Add extensible enum as response in FirstTest-Cadl project, and this method is not chosen to generate convenience method, so the enum will not be generated.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**